packages/common/src/hardhat/tasks/xchainSetup.ts

```diff
@@ -166,41 +202,75 @@ task("setup-l1-arbitrum-xchain", "Configures L1 cross chain smart contracts for
-    ]);
-
-    // Submit Finder transactions:
-    assert(finderOwner === deployer, `Accounts[0] (${deployer}) is not equal to finder owner (${finderOwner})`);
-    if (finderChildMessenger !== ChildMessenger.address) {
-      console.log(`Setting finder ChildMessenger to ${ChildMessenger.address}...`);
-      const setMessengerTxn = await finder.methods
-        .changeImplementationAddress(utf8ToHex("ChildMessenger"), ChildMessenger.address)
-        .send({ from: deployer });
-      console.log(`...txn: ${setMessengerTxn.transactionHash}`);
-    }
-    if (finderRegistry !== Registry.address) {
-      console.log(`Setting finder Registry to ${Registry.address}...`);
-      const setRegistryTxn = await finder.methods
-        .changeImplementationAddress(utf8ToHex("Registry"), Registry.address)
-        .send({ from: deployer });
-      console.log(`...txn: ${setRegistryTxn.transactionHash}`);
-    }
+  const ParentMessenger = await deployments.get("Boba_ParentMessenger");
+  const messenger = new web3.eth.Contract(ParentMessenger.abi, ParentMessenger.address);
+  const OracleHub = await deployments.get("OracleHub");
+  const oracleHub = new web3.eth.Contract(OracleHub.abi, OracleHub.address);
+  const GovernorHub = await deployments.get("GovernorHub");
+  const governorHub = new web3.eth.Contract(GovernorHub.abi, GovernorHub.address);
+
+  console.log(`Found ParentMessenger @ ${messenger.options.address}`);
+  console.log(`Found OracleHub @ ${oracleHub.options.address}`);
+  console.log(`Found GovernorHub @ ${governorHub.options.address}`);
+
+  const OracleSpoke = await companionNetworks.boba.deployments.get("OracleSpoke");
+  const ChildMessenger = await companionNetworks.boba.deployments.get("Boba_ChildMessenger");
+  const GovernorSpoke = await companionNetworks.boba.deployments.get("GovernorSpoke");
+
+  await setupParentMessenger(messenger, deployer, ChildMessenger, OracleHub, GovernorHub, OracleSpoke, GovernorSpoke);
+
+  // Submit parent messenger cross-chain transactions:
+  console.log(`Setting child oracle spoke address to ${OracleSpoke.address}...`);
+  const setChildOracleSpokeTxn = await messenger.methods
+    .setChildOracleSpoke(OracleSpoke.address)
+    .send({ from: deployer });
+  console.log(`...txn: ${setChildOracleSpokeTxn.transactionHash}`);
+  console.log(`Setting child parent messenger to ${messenger.options.address}...`);
+  const setChildParentMessengerTxn = await messenger.methods
+    .setChildParentMessenger(messenger.options.address)
+    .send({ from: deployer });
+  console.log(`...txn: ${setChildParentMessengerTxn.transactionHash}`);
+
+  // Submit oracle hub transactions:
+  await setupHub(oracleHub, deployer, messenger.options.address, 288);
+
+  // Submit governor hub transactions:
+  await setupHub(governorHub, deployer, messenger.options.address, 288);
+});
+
+task("setup-l2-xchain", "Configures L2 cross chain smart contracts for Arbitrum bridge").setAction(async function (
```

Comment:
nit: this works for chains other than arbitrum, right?

Reply:
Oh true, will fix comment